core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -34,20 +36,14 @@ sealed trait Chunk[+A] { self =>
    */
   val length: Int
 
-  /**
-   * Returns the concatenation of this chunk with the specified chunk.
-   */
-  final def ++[A1 >: A](that: Chunk[A1]): Chunk[A1] =
```

Comment:
Can we keep this on `Chunk` and override it in `NonEmpty` instead of using the implicit syntax class? I think that will be more discoverable and looks like it improves type inference in some cases (e.g. `++` in `Sink.scala` below).

Reply:
It is not possible to keep in on `Chunk` without losing some of the new features around concatenation.

```
**Chunk ++ NonEmpty => NonEmpty**
NonEmpty ++ Chunk => NonEmpty
NonEmpty ++ NonEmpty => NonEmpty
```

If I put them back
```scala 
scala> Chunk(1,2,3) ++ Chunk(1)
<console>:25: error: ambiguous reference to overloaded definition,
both method ++ in trait NonEmpty of type [A1 >: Int](that: zio.Chunk[A1])zio.Chunk.NonEmpty[A1]
and  method ++ in trait Chunk of type [A1 >: Int](that: zio.Chunk.NonEmpty[A1])zio.Chunk.NonEmpty[A1]
match argument types (zio.Chunk.NonEmpty[Int])
       Chunk(1,2,3) ++ Chunk(1)
```

This still works
```
scala> Chunk.empty ++ Chunk(1)
res1: zio.Chunk.NonEmpty[Int] = Singleton(1)
```

If `def ++[A1 >: A](that: NonEmpty[A1]): NonEmpty[A1]`  is removed, then it works again.

Reply:
Solved it adding covariance in ChunkOps
```scala

implicit class ChunkOps[+A](private val self: Chunk[A]) extends AnyVal {
    /**
     * Returns the concatenation of this chunk with the specified chunk.
     */
    final def ++[A1 >: A](that: Chunk[A1]): Chunk[A1] = concat(self, that)

    /**
     * Returns the concatenation of this chunk with the specified chunk.
     */
    final def ++[A1 >: A](that: NonEmpty[A1]): NonEmpty[A1] = concat(self, that)
  }
```

Unchanging the code on Sink.